extensions/src/main/java/org/apache/falcon/extensions/Extension.java

```diff
@@ -92,11 +104,21 @@ private static String getWFPath(final Map<String, String> extensionResources,
         /* Get Lib path */
         String wfLibPath = store.getExtensionLibPath(name);
         Entity entity = ExtensionProcessBuilderUtils.createProcessFromTemplate(extensionTemplate,
-                name, extensionProperties, wfPath, wfLibPath);
+                name, configProperties, wfPath, wfLibPath);
         if (entity == null) {
             throw new FalconException("Entity created from the extension template cannot be null");
         }
         LOG.info("Extension processing complete");
-        return Arrays.asList(entity);
+        return Collections.singletonList(entity);
+    }
+
+    @Override
+    public void validateExtension(String extensionName, InputStream extensionConfigStream) throws FalconException {
+        throw new UnsupportedException("Not yet Implemented");
```

Comment:
Apologies again for not pointing out earlier. Why unsupported? This could invoke org.apache.falcon.extensions.Extension#validateProperties. isn't it?